[WIP] Costco store vault

unid15 opened the pull request:
![](http://puu.sh/nP33a/db11624f0e.jpg)

Added a huge store that can appear somewhere in space. It's operated by robots! 
In the store you can buy expensive things for cheap, and very rare things (switchtools, battering rams, tank transfer valves, ...) for lots of money. You can also buy circuitboards (75$ each) or clothing (150$ each). Stock is randomly generated every round

How to buy something:
![](http://puu.sh/nNNel/7b03770372.gif)

It's guarded by lots of extremely robust and fast robots with 200 health, built-in rifle and 15-25 melee damage. Crossing the warning stripes while holding an unpaid item will [instantly lock down the entire store and send the guard bots after you](http://puu.sh/nNMxh/9a25a8f05a.jpg). 

If you don't have any money, you can get a drink at the bistro - alcohol and water is free there.

OTHER CHANGES:
- IDs on corpse landmarks now properly shows occupation
- existing_typesof now performs an additional check

TODO:
- Balance the loot and prices perhaps
- Look for bugs

ihadtoregisterforthis commented:
DEEPEST LORE

Intigracy commented:
How about adding a table that continually respawns 1 nutriment "samples" after one is picked up.

Also, when the store goes into lockdown, does everyone inside get affected?

Iamgoofball commented:
welcom to cosco

i lov u

Intigracy commented on vgstation13.dme:
```diff
@@ -1933,6 +1933,7 @@
 #include "maps\tgstation.dm"
 #include "maps\defficiency\areas.dm"
 #include "maps\defficiency\pipes.dm"
+#include "maps\randomvaults\costco.dm"
```
bad dog

unid15 commented:
Everybody is killed, these robots don't take shit from anybody (I should probably change that, but it'd be difficult trying to determine the offender when, for example, an unpaid item is put in a crate and pushed out of the store)

Intigracy commented:
Nah just have them kill everyone.

Does the store ever stop being in lockdown?

unid15 commented:
Nah

You're stuck in there forever unless you've got tools to deconstruct 2 rwalls

Imagebard commented:
>Nah

Can't you just make the robots go back to their default positions and open everything back once everyone inside is dead?

Imagebard commented:
Wait a minute
You're bolting people to the 1-tile big toilets too?

icantthinkofanameritenow commented on maps/randomvaults/costco.dm:
```diff
@@ -0,0 +1,396 @@
+
+/mob/living/simple_animal/hostile/retaliate/costco_guardian
+	name = "Costco MERC-Bot"
+	desc = "Equipped with a ballistic weapon and a melee range shocker that is powerful enough to knock out a mega goliath through three layers of protection, this EMP-proof bot is not to be messed around with."
```
That reminds me, is anyone still working on that mega-goliath?

icantthinkofanameritenow commented:
If you can make a way to de-lockdown the place when the offender is dead, it should be good.

ihadtoregisterforthis commented:
Is the stock randomized?
Why do those trees have snow while the grass doesn't?
Is it possible to have a small magazine with the prices?
Can I steal the bootyborgs?
What role does the baby hold?

Color a few tiles so it doesn't look so plain. Unless Costcos are that way.

Shadowmech88 commented:
Make sure to account for people using telescience/portals to steal things without taking them past the warning line.

SkowronX commented on maps/randomvaults/costco.dm:
```diff
@@ -0,0 +1,396 @@
+
+	icon = 'icons/mob/robots.dmi'
+	icon_state = "booty-red"
+
```
You will regret this.

Intigracy replied:
nah

clusterfack commented on maps/randomvaults/costco.dm:
```diff
@@ -0,0 +1,396 @@
+	say("Vaae-#@ism is forbieeen at aostAo. Your aAAount will be terminated. Aonnection to aostAo mainframe lost... Self-destructing in 5... 4... 3...")
+	say("2... 1...")
+
+	spawn(3)
```
spawn(3 SECONDS)

clusterfack commented on maps/randomvaults/costco.dm:
```diff
@@ -0,0 +1,396 @@
+	icon = 'icons/obj/weapons.dmi'
+	icon_state = "toddler"
+
+	invisibility = 0
```
Use one of the defines for this

unid15 commented:
Stock is randomized (items are spawned from a pool). Also the shopping area is telescience-proof so don't worry about that

Updated the map. Robots now say "Welcome to Costco I love you", there's also a robot that can recite Costco's rules to you. Breaking a wall, window, structure or destroying an item now triggers the alarm and lockdown. Added a little maintenance area with cyborg rechargers, crematorium and a prison cell for a captive roboticist (who committed suicide, unfortunately for the robots). Also made the store more secure, with 3 layer walls - 1st layer is rwall with plasma glass covering, 2nd layer is rwall covered with reinforced girders and the 3rd layer is an ordinary wall. Will fix the trees later

![](http://puu.sh/nP33a/db11624f0e.jpg)

Intigracy commented:
Does preventing telescience from reaching it also prevent you using the hand tele?

Shadowmech88 commented:
Is it 100% telescience proof? Even with a bluespace crystal in the telepad?

unid15 commented:
Shopping area is 100% telescienceproof, other areas require a crystal. Hand tele currently works but I'm going to fix it

Costco now features food samples! (thanks for the idea @Intigracy)
![samples](https://cloud.githubusercontent.com/assets/9512290/13964131/9c9644e8-f068-11e5-891f-74aa506128d6.gif)

SkowronX commented:
:-1:  unless you replace borg sprites.

Imagebard commented:
Skowron have you never heard of "Sex sells"?

Intigracy commented:
There's nothing wrong with the borg sprite.

gbasood commented:
Borg sprites are fine.

Imagebard commented:
>Even if shitters had the sprite it would be fine.

I'd politely disagree, but Iasking if this can be merged is more important right now.

Imagebard commented:
Oh fuck its the !!CONFLICTS!!

unid15 commented:
there must be a way to add new vaults without causing conflicts with everybody else

Intigracy commented:
There is if you put the defines on different lines than other people, I
think

PJB3005 commented:
@unid15 could make vaults be loaded from external files at runtime but the issue is more complicated ones like this PR are gonna be fucked nontheless.

Intigracy commented:
As you can see, my PR isn't conflicting.

[8951]

Putting your defines at the end of the list like everyone else is your problem.

Intigracy commented:
Also you still need to remove your map file from the DM's include.

unid15 commented:
Turned the outer walls into reinforced walls (because it's FAR too easy to steal items otherwise, even with 3 layered rwalls + rgirders). Added clothing (150$ each) and circuits (75$ each) sections to the store. All types of clothing and circuits can spawn there (except for the OP ones).

![](http://puu.sh/nTnFA/e4e973e13c.png) ![](http://puu.sh/nTooy/b4add6030c.jpg)

Gonna stop adding content now and focus on testing and bugfixing

Intigracy commented:
This branch has conflicts that must be resolved

unid15 commented:
>[WIP]

Intigracy commented:
:^)

F-OS commented:
what is with the invulnerable walls, people should be able to break out some way.

Intigracy commented:
None of the walls are invulnerable?

F-OS commented on maps/randomvaults/costco.dmm:
```diff
@@ -1,176 +1,227 @@
-"ad" = (/turf/simulated/wall/r_wall,/area/vault/supermarket/entrance)
-"ae" = (/obj/structure/window/reinforced/plasma,/turf/simulated/wall/r_wall,/area/vault/supermarket/entrance)
-"af" = (/obj/structure/window/reinforced/plasma{dir = 4},/turf/simulated/wall/r_wall,/area/vault/supermarket/entrance)
+"ad" = (/turf/simulated/wall/invulnerable,/area/vault/supermarket/entrance)
```
that is the invulnerable wall

Imagebard commented:
@Intigracy in the last image unid posted, the inner walls in the store are the same as the ones in CentComm.

unid15 commented:
@F-OS, it's far too easy to shoplift without invulnerable walls by standing next to an outer wall, putting walls/girders/any dense objects around you and slowly digging your way out

PJB3005 commented:
Also unid correct me if I'm wrong but if you do trigger lock down and survive you can still go through the airlock, right?

Probe1 commented:
Too literal, not SS13.  Rename it to something in-lore like Donk-Mart or whatever you particularly feel like.  There's actually a lot of well written lore hidden in an away mission file somewhere that Cael never finished coding.  

Anything in-universe except fuckin costco.

NigglyWiggly commented:
You can use one of my booty borg sprites if you don't wanna have twig-arms as a cashier

Also yeah change the name

FudgePucker commented:
If the lockdown is too hard to implement and have it go down, maybe just have it gib shoplifters.

Also maybe put in a teleporter

Shadowmech88 commented:
W H E N
H
E
N

ComicIronic commented on code/defines/obj/weapon.dm:
```diff
@@ -390,6 +390,10 @@
 	var/armed = 0
 	var/obj/item/weapon/grenade/iedcasing/IED = null
 
+/obj/item/weapon/legcuffs/beartrap/primed
```
/armed, for consistency.

ihadtoregisterforthis commented:
Sometime someday

unid15 commented:
Hey that's funny, I was just thinking about reviving this as an away mission, instead of a vault

Probe1 commented:
that'd be neat.  I still like the idea as long as the naming convention fits SS13

PJB3005 commented:
This is basically dead, RIP.

Intigracy commented:
RIP

FudgePucker commented:
RIP
press f to pay respects.
